### ISSUE TITLE
Progress bar with multiple elements

### ISSUE THREAD

**Issue by N3XT191** [palantir/blueprint#4490]
#### Environment

- __Package version(s)__: core 3.37
- __Browser and OS versions__: Safari 14, macOS BigSur

#### Feature request

Either extend the current progress bar or create a "MultiProgressBar" component that allows setting multiple elements in different colors/intents that are stacked on top of each other.

#### Examples

![alt](https://raw.githubusercontent.com/N3XT191/react-multi-progress/main/docs/progressbar.png)

I made [this](https://www.npmjs.com/package/react-multi-progress) for a project that mostly uses blueprint and thought it would be nice to have in blueprint too. 

I am willing to write the code for this but would need some guidance on how to structure it (adjust ProgressBar or make MultiProgressBar, prop structure etc)

**Comment by N3XT191** [comment]
<img width="821" alt="Screenshot 2021-01-21 at 14 20 17" src="https://user-images.githubusercontent.com/4399535/105356564-dabe0980-5bf3-11eb-8b7f-b49cb968dd65.png">

In case anyone is wondering for what this is useful, here is the main place it's used in my project

**Comment by adidahiya** [comment]
Thanks for the additional context. It looks like this kind of component exists in other libraries such as [Fomantic UI](https://fomantic-ui.com/modules/progress.html#multiple-bars). I would be open to adding it to Blueprint. We have something similar in MultiSlider, where we render a bar with multiple segments. You could probably borrow/copy some of the MultiSlider CSS and prop interface design to build a MultiProgressBar.

**Comment by N3XT191** [comment]
For Syntax, the following would probably be easiest to implement:

```jsx
<MultiProgressBar values={[0.1,0.5,1.0]} intents={[Intent.SUCCESS,Intent.WARNING,Intent.ERROR]}/>
```

but this would be most similar to the MultiSlider syntax

```jsx
<MultiProgressBar >
    <MultiProgressBarElement value={0.1} intent={Intent.SUCCESS}/>
    <MultiProgressBarElement value={0.5} intent={Intent.WARNING}/>
    <MultiProgressBarElement value={1.0} intent={Intent.ERROR}/>
</MultiProgressBar >
```

What might be best would be to do the following, but I don't know how to make the stacking so all 3 progress meters are visible (and not blocked by the other backgrounds)

```jsx
<MultiProgressBar >
    <ProgressBar value={0.1} intent={Intent.SUCCESS}/>
    <ProgressBar value={0.5} intent={Intent.WARNING}/>
    <ProgressBar value={1.0} intent={Intent.ERROR}/>
</MultiProgressBar >
```

Thoughts?
